draft-ietf-tls-tls13.md

```diff
@@ -1908,33 +1866,25 @@ client_shares
   in the "supported_groups" extension.
 
 server_share
-: A single KeyShareEntry value for the negotiated cipher suite.
+: A single KeyShareEntry value that is in the same group as one of the
+  client's shares.
 {:br }
 
 Clients offer an arbitrary number of KeyShareEntry values, each
 representing a single set of key exchange parameters. For instance, a
 client might offer shares for several elliptic curves or multiple
 FFDHE groups.  The key_exchange values for each KeyShareEntry MUST by
 generated independently.  Clients MUST NOT offer multiple
-KeyShareEntry values for the same group.  Clients MUST NOT offer
-any KeyShareEntry values for groups not listed in the client's
-"supported_groups" extension.
-
-Servers offer exactly one KeyShareEntry value, which corresponds to
-the key exchange used for the negotiated cipher suite.  Servers MUST
-NOT offer a KeyShareEntry value for a group not offered by the client
-in its corresponding KeyShare or "supported_groups" extension.
-
-Implementations MAY check for violations of these rules and
-and MAY abort the connection with a fatal "illegal_parameter" alert
-if one is violated.
-
-If the server selects an (EC)DHE cipher suite and no mutually
-supported group is available between the two endpoints' KeyShare
-offers, yet there is a mutually supported group that can be found via
-the "supported_groups" extension, then the server MUST reply with a
-HelloRetryRequest.  If there is no mutually supported group at all,
-the server MUST NOT negotiate an (EC)DHE cipher suite.
+KeyShareEntry values for the same group.  Clients MUST NOT offer any
+KeyShareEntry values for groups not listed in the client's
+"supported_groups" extension.  Servers MAY check for violations of
+these rules and and MAY abort the connection with a fatal
+"illegal_parameter" alert if one is violated.
+
+Servers offer exactly one KeyShareEntry which corresponds to one of
+the KeyShareEntries offered by the client. Servers MUST NOT send
```

Comment:
This particular wording makes it sound unimportant, like the server should pick one at random to reply to.
Maybe "[...] which corresponds to one of the KeyShareEntries offered by the client, that the server has selected for the negotiated key exchange."